lib/handler/decorator-handler.js

```diff
@@ -14,12 +26,88 @@ module.exports = class DecoratorHandler {
     this.#handler = handler
   }
 
+  // New API
+
+  onRequestStart (abort) {
+    this.#handler.onRequestStart?.(abort)
+    this.#handler.onConnect?.(abort)
+  }
+
+  onResponseStart (resume) {
+    let ret = true
+
+    if (this.#handler.onResponseStart?.(resume) === false) {
+      ret = false
+    }
+
+    this.#resume = resume
+
+    return ret
+  }
+
+  onResponseHeaders (headers, statusCode) {
+    let ret = true
+
+    if (this.#handler.onResponseHeaders?.(headers, statusCode) === false) {
+      ret = false
+    }
+
+    if (this.#handler.onHeaders) {
+      const rawHeaders = toRawHeaders(headers)
+      if (this.#handler.onHeaders(statusCode, rawHeaders, this.#resume) === false) {
+        ret = false
+      }
+    }
```

Comment:
Here we will have consistency conditions, isn't it?
Because if one of them returns `false`, and other `true`, we will continue reading chunks, causing that possibly one of them gets corrupted.

Shall we better allow one or the other but nut the two of them?

Reply:
Not sure I follow. If one of them returns false then the return value is false.

Reply:
Sure. But then what would happen if e.g. `onResponseHeaders` returns `false` but `onHeaders` returns `true`?

I think we are allowing them to pass both APIs at the same time? Wouldn't that maybe cause inconsistencies?

Reply:
No that's fine, if one returns false then it's paused.